seabird.proto

```diff
@@ -4,168 +4,163 @@ option go_package = ".;pb";
+  //
+  // NOTE: help is a reserved command and cannot be registered by plugins.
   map<string, CommandMetadata> commands = 2;
 }
 
+// CommandMetadata groups together a command's name along with short in-line
+// help and the full private help.
 message CommandMetadata {
   string name = 1;
   string short_help = 2;
   string full_help = 3;
 }
 
-message Event {
-  oneof inner {
-    MessageEvent message = 1;
-    PrivateMessageEvent private_message = 2;
-    MentionEvent mention = 3;
-    CommandEvent command = 4;
-  }
-}
-
-message MessageEvent {
-  string reply_to = 1;
-  string sender = 2;
-  string message = 3;
-}
-
-message PrivateMessageEvent {
-  string reply_to = 1;
-  string sender = 2;
-  string message = 3;
-}
-
-message MentionEvent {
-  string reply_to = 1;
-  string sender = 2;
-  string message = 3;
-}
-
-message CommandEvent {
-  string reply_to = 1;
-  string sender = 2;
-  string command = 3;
-  string arg = 4;
-}
+// Requests a plugin can make
 
+// Send a message to a given a channel.
 message SendMessageRequest {
-  Identity identity = 1;
+  common.Identity identity = 1;
 
-  string target = 2;
-  string message = 3;
+  string channel_id = 2;
+  string text = 3;
 }
 
 message SendMessageResponse {}
 
-message ListChannelsRequest {
-  Identity identity = 1;
-}
+// Send a private message to a given user.
+message SendPrivateMessageRequest {
+  common.Identity identity = 1;
 
-message ListChannelsResponse {
-  repeated string names = 1;
+  string user_id = 2;
+  string text = 3;
 }
 
-message ChannelInfoRequest {
-  Identity identity = 1;
+message SendPrivateMessageResponse {}
 
-  string name = 2;
-}
+// Request to join a channel.
+message JoinChannelRequest {
+  common.Identity identity = 1;
+
+  string server_id = 2;
 
-message User {
-  string nick = 1;
+  // NOTE: this channel_name is the only place name is used for an identifier -
+  string channel_name = 3;
 }
 
-message SetChannelTopicRequest {
-  Identity identity = 1;
+message JoinChannelResponse {}
 
-  string name = 2;
-  string topic = 3;
+// Request to leave a channel.
+message LeaveChannelRequest {
+  common.Identity identity = 1;
+
+  string channel_id = 2;
+  string exit_message = 3;
 }
 
-message SetChannelTopicResponse {}
+message LeaveChannelResponse {}
 
-message ChannelInfoResponse {
-  string name = 1;
-  string topic = 2;
-  repeated User users = 3;
+// Request to list all connected backends.
+message ListBackendsRequest {
+  common.Identity identity = 1;
 }
 
-message JoinChannelRequest {
-  Identity identity = 1;
-
-  string name = 2;
+message ListBackendsResponse {
+  repeated common.Backend backends = 1;
 }
 
-message JoinChannelResponse {}
+// Get info about a specific connected backend.
+message BackendInfoRequest {
+  common.Identity identity = 1;
 
-message LeaveChannelRequest {
-  Identity identity = 1;
+  string backend_id = 2;
+}
 
-  string name = 2;
-  string message = 3;
+message BackendInfoResponse {
+  common.Backend service = 1;
 }
```

Comment:
Should this be a shallow list of channel IDs too or is that too bulky?

Reply:
I'm going to punt on this because it can be added later.